src/adapters/satoshi-staking.adapter.ts

```diff
@@ -0,0 +1,39 @@
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys.js';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking.js';
+// Use the full path to import instead of `../types`
+import {Coco__factory} from '../types/factories/Coco__factory.js';
+
+@injectable(
+  {
+    scope: BindingScope.SINGLETON, // Mark the adapter as a singleton
+  },
+  // Mark it as an extension to staking contracts service
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class SatoshiStakingContractAdapter extends BaseStakingContractAdapter {
+  /**
+   * The contract address
+   */
+  contractAddress = '0xeDFdDb8eA41DEb967E75dc14Ff56A821B1Ef3306';
+
+  /**
+   * Assets that can be staked to this contract (satoshi nft)
+   */
+  supportedAssets: StakingAsset[] = [
+    {
+      asset: 'ERC721:0x978f7f915eB562d3819a82Daf4CEe01dD636C2D1',
+    },
+  ];
+
+  /**
+   * Get staked token ids for the given owner
+   * @param owner - Owner address
+   * @returns
+   */
+  getStakedTokenIds(owner: string): Promise<BigNumber[]> {
+    const contract = Coco__factory.connect(this.contractAddress, this.provider);
```

Comment:
Should it be `SatoshiStaking__factory`?

Reply:
@raymondfeng Ooops thanks for spotting that. Question: the getStakedTokenIds() must return staked NFT token IDs from owner?

Reply:
Yes

Reply:
oh ok thanks @raymondfeng I'm checking if the contract has such function, at a glance it doesn't (it was written a while ago)

Reply:
hey @raymondfeng questions:
1. Does this adapter run on the EVM? I can't do an http call inside getStakedTokenIds() right?
2. We also have an ERN fixed staking contract, that one will work - should I do a new pull request or just include it in the current one?
Thanks!

Reply:
Please fix the issue as I commented.

Reply:
hey @raymondfeng 
I changed the function so it calls the endpoint instead of a contract function. Questions:
1. The tests are failing for me
```
  32) gets staked token balances for FlooringProtocol:
     Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (transaction={"to":"0x49AD262C49C7aA708Cc2DF262eD53B64A17Dd5EE","data":"0xe89a7aed","accessList":null}, code=CALL_EXCEPTION, version=providers/5.7.1)

```     
is this something sporadic or failing from my machine?

2. Is /src/types/SatoshiStaking.ts still needed? (since I am not calling the contract anymore from the adapter)

Thanks!

Reply:
You can remove the abi json.